src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeModule.cs

```diff
@@ -199,10 +212,13 @@ public override byte[] ResolveSignature(int metadataToken)
                                 nameof(metadataToken));
                     }
 
-                    fieldHandle = ModuleHandle.ResolveFieldHandleInternal(GetNativeHandle(), tk, typeArgs, methodArgs);
+                    fieldHandle = moduleHandle.ResolveFieldHandle(tk, typeArgs, methodArgs).GetRuntimeFieldInfo();
```

Comment:
These two ResolveFieldHandle calls take exactly same arguments. Should we have just common one after the `if { .. }` ?

Reply:
Not sure about this. Previously the second wasn't in the if; and then the `fieldHandle` was unconditionally overwritten, so the bit in the `if` never produced a result... 🤔

Reply:
Other than the additional `throw`s

Reply:
Ah I see what its doing